src/python/tests/core/bot/minimizer/js_minimizer_test.py

```diff
@@ -0,0 +1,171 @@
+      tests would have been run to validate that the correct tokens will be
+      removed."""
+    txt = ''
+    for index in hypothesis:
+      txt += testcase.tokens[index]
+
+    self._hypotheses_tested.append(txt)
+
+  def _mock_test_function(self):
+    return True
+
+  def test_minimize_empty_string(self):
+    """Minimizer does not break on empty data."""
+    data = ''
+
+    self._minimizer.minimize(data)
+
+    self.assertEqual(self._hypotheses_tested, [])
+
+  def test_if_hypothesis(self):
+    """e.g.: if (statement_that_evaluates_to_true) { crash() } -> crash()."""
+    data = "if(boolean) { crash }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("if(boolean) {}", self._hypotheses_tested)
+
+  def test_try_catch_hypothesis(self):
+    """e.g.: try { crash() } catch(e) {} -> crash()."""
+    data = "try{ crash() } catch(e){ }"
+    self._minimizer.minimize(data)
+
+    self.assertIn("try{} catch(e){ }", self._hypotheses_tested)
+
+  def test_handle_if_else(self):
+    """Make sure the minimizer runs all of the other hypothesis cleanly on
+      if else."""
+    data = "if(boolean) {crash} else { do_something_else }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("if(boolean) {}", self._hypotheses_tested)
+    self.assertIn("if(boolean) {} else { do_something_else }",
+                  self._hypotheses_tested)
+    self.assertIn("if(boolean) {crash} else {}", self._hypotheses_tested)
+
+  def test_remove_function_call(self):
+    """Test that it removes functions calls effectively."""
+    data = "function name(param1, param2){ stuff inside function }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("function name(param1, param2){}", self._hypotheses_tested)
+
+  def test_handle_bracket_with_new_line(self):
+    """Test for try/catch with extra whitespace."""
+    data = """try{\n\tcrash()\n}\ncatch(e){\n\n}"""
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("try{}", self._hypotheses_tested)
+    self.assertIn("try{}\ncatch(e){\n\n}", self._hypotheses_tested)
+
+  def test_handle_function_with_new_lines(self):
+    """Test for functions with extra whitespace."""
+    data = "function name(param1,\n\t\tparam2){\n\tstuff inside function\n}"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("function name(param1,\n\t\tparam2){}",
+                  self._hypotheses_tested)
+
+  def test_remove_outer_paren(self):
+    """e.g.: assertTrue(crash()); -> crash()."""
+    data = "assertTrue(crash());"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("assertTrue()", self._hypotheses_tested)
+
+  def test_remove_inside_paren(self):
+    """Hypothesis is Remove everything between the parentheses.
+      e.g.: crash(junk, more_junk) -> crash()."""
+    data = "crash(junk, more_junk)"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("junk, more_junk", self._hypotheses_tested)
+
+  def test_remove_paren_to_start_of_line(self):
+    """e.g.: leftover_junk = (function() {
```

Comment:
Nit: some high-level description for these before each of the examples.

Reply:
Done